### ISSUE TITLE
[Question] about setting of threshold

### ISSUE THREAD

**Issue by Ronalmoo** [mdabashar/TAnoGAN#1]
Hello, I am a graduate student.
First of all, thank you for sharing the code. 
And after reading your paper, I have a question about how to set up a threshold. 

I'd appreciate your reply. 
Thank you.

**Comment by izmendi** [comment]
Yes, the value of the threshold is what delimits which instances are outliers and which are not. In the code it appears as a prefixed value, but perhaps (this is my point of view) the most appropriate would be an optimization algorithm that decides the optimal value of the threshold. For this, it would be necessary to work with a supervised dataset ..... but TAnoGAN is unsupervised, in this case I don't know what would be the optimal way to work.

But, if TAnoGAN es unsupervised, the Generator is also training with outlier values, how is it posible? TAnoGAN will be learning also from abnormal behaviours and will not be able to distinguish them in the validation phase ......
